jscomp/ext/bs_hash_stubs.ml

```diff
@@ -0,0 +1,13 @@
+let hash_string : string -> int = Hashtbl.hash
```

Comment:
This is prob the most controversial change. It means that C-written functions like `caml_bs_hash_int` are replaced with OCaml counterparts. If this PR goes through, I can remove `bs_hash_stubs` and the corresponding C files in another PR.